src/index.ts

```diff
@@ -362,15 +375,21 @@ function floatShifts(
     }
 
     const shift = xShift > yShift ? xShift : yShift
-    let moreS = 0
-    if (typeof moreShift === 'number') {
-      moreS = moreShift
-    }
 
-    x = addZeros(x.replace('.', ''), shift + moreS - xShift)
+    x = addZeros(x.replace('.', ''), shift - xShift)
     y = addZeros(y.replace('.', ''), shift - yShift)
 
-    const out: ShiftPair = { x, y, shift }
+    let extraShift = 0
+    if (doFloat) {
```

Comment:
Add a comment explaining doFloat and it's effects either here or in the function's description.

Reply:
done